src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -53,27 +53,40 @@ public void Write()
 
             this.emitter.EmitProperty("contentVersion", "1.0.0.0");
 
-            writer.WritePropertyName("parameters");
-            this.EmitParameters();
-            
-            writer.WritePropertyName("functions");
-            writer.WriteStartArray();
-            writer.WriteEndArray();
+            this.EmitParametersIfPresent();
+
+            this.EmitFunctionsIfPresent();
 
-            writer.WritePropertyName("variables");
-            this.EmitVariables();
+            this.EmitVariablesIfPresent();
 
-            writer.WritePropertyName("resources");
-            this.EmitResources();
-
-            writer.WritePropertyName("outputs");
-            this.EmitOutputs();
+            this.EmitResources();
+
+            this.EmitOutputsIfPresent();
 
             writer.WriteEndObject();
-        }
-
-        private void EmitParameters()
+        }
+
+        private void EmitFunctionsIfPresent()

```

Comment:
You should just remove the function. Whoever ends up adding user-defined function support will bring it back.

Reply:
Done.